### ISSUE TITLE
Can the system be trained using RNA-Seq data?

### ISSUE THREAD

**Issue by melkerdawy** [google/deepvariant#115]
I did not find any mention of the possibility of using RNA-Seq data in training mode.
Is this possible but you choose to focus on DNA data only or there are constraints in the model that need to be modified before training deepvariant  on RNA variant calling.

**Comment by pichuan** [comment]
Hi @melkerdawy ,
the DeepVariant codebase is currently designed to for DNA data only. The underlying tool and principle (of converting genomic data into a machine learning problem) could be generalized. But the existing tool as is isn't designed or used for RNA-seq data.

In another word - it could work, but it will be open-ended research. I'd recommend you looking into how DeepVariant is done, and look into the [Nucleus](https://github.com/google/nucleus) library as well. We just recently announced a pip package for Nucleus. 

Feel free to share your experimental results and discuss any issues you've encountered. We'll try our best to answer and discuss with you here.

(Closing for now. Feel free to re-open)

**Comment by AldoCP** [comment]
Sort of a follow-up question, on a related application: can DeepVariant take an RNA-seq bam file obtained via GATK's best practices (link [here](https://software.broadinstitute.org/gatk/documentation/article.php?id=3891)), and get a trustworthy output? 

A quick check on a specific locus indicates that DeepVariant's WGS applied to RNA-seq retrieves a few more variants than GATK's "HaplotypeCaller". Can we trust it? Has anybody done more comprehensive testing of this?

**Comment by pichuan** [comment]
Adding @AndrewCarroll to see if he has more thoughts.
I suspect if we specifically train an rna seq data with the same underlying algorithm, it will work well. However, we have not yet done anything on RNA data like I mentioned in my previous comment.

**Comment by AndrewCarroll** [comment]
Hi @AldoCP,

We have not comprehensively investigated performance on RNA-Seq samples yet. Because some components of the RNA-Seq problem more closely resemble variant calling in DNA-Seq exomes, I recommend that you use the exome model instead of the WGS one.

A proper investigation would probably involve looking at the variant calls that result from the same sample with both DNA-Seq and RNA-Seq and investigating the concordance of calls. I expect that sort of comparison is probably worth a paper. 

In general, we observe that when DeepVariant is applied in other domains beyond its training data, it tends to undercall as opposed to call false positive events, so if you are getting calls in DeepVariant and not HaplotypeCaller, those are worth looking into. If you do so, we would be interested in your feedback.

**Comment by AldoCP** [comment]
Thanks for the feedback. 

I went back to my files and just realized that my previous comment was inaccurate: the locus I analyzed on RNASeq was "chr20:10,000,000-10,040,000"; an exonic variant (chr20:10019093) was detected by both GATK and DeepVariant (WGS model). 

As mentioned, I didn't do extensive tests at all (it was just that one locus) -- I'm happy to do further analysis if relevant,

**Comment by AldoCP** [comment]
Thanks for the feedback.

I went back to my files and just realized that my previous comment was inaccurate: the locus I analyzed on RNASeq was "chr20:10,000,000-10,040,000"; the same exonic variant (chr20:10019093) was detected by both GATK and DeepVariant (WGS model) in my sample.

As mentioned, I didn't do extensive tests at all (it was just that one locus) -- I'm happy to do further analysis if relevant,

**Comment by danielecook** [comment]
@AldoCP @melkerdawy although this was closed some time ago, I have an update related to this question.

We have successfully trained and released an Illumina RNA-seq model and [case study](https://github.com/google/deepvariant/blob/r1.4/docs/deepvariant-rnaseq-case-study.md). 

DeepVariant now has a flag (`--split_skip_reads`) that can be used in conjunction with training or inference with RNA-seq data. The flag is required to efficiently process RNA-seq data.
